pkg/services/ngalert/services/alertmanager_config.go

```diff
@@ -0,0 +1,237 @@
+package services
+
+import (
+	"context"
+	"encoding/base64"
+	"errors"
+	"fmt"
+
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/services/ngalert/api/tooling/definitions"
+	"github.com/grafana/grafana/pkg/services/ngalert/models"
+	"github.com/grafana/grafana/pkg/services/ngalert/notifier"
+	"github.com/grafana/grafana/pkg/services/ngalert/store"
+	"github.com/grafana/grafana/pkg/services/secrets"
+)
+
+type UnknownReceiverError struct {
+	UID string
+}
+
+func (e UnknownReceiverError) Error() string {
+	return fmt.Sprintf("unknown receiver: %s", e.UID)
+}
+
+type AlertmanagerConfigRejectedError struct {
+	Inner error
+}
+
+func (e AlertmanagerConfigRejectedError) Error() string {
+	return fmt.Sprintf("failed to save and apply Alertmanager configuration: %s", e.Inner.Error())
+}
+
+// AlertmanagerConfigService is a domain-layer service which manages configs for the Grafana alertmanager.
+type AlertmanagerConfigService struct {
+	mam     *notifier.MultiOrgAlertmanager
```

Comment:
The initial issue I ran into here was that `alertmanager_config.go` now is moved into the `notifier` package following the other comment. So, its use of the MAM/store/secret service is now internal to the package - it doesn't need to depend on a narrow public API exported by `notifier`.

Firstly, I broke the encryption code out into `encryption.go` almost verbatim as what was described. I like the result of this a lot, since that logic should never have been bound to `alertmanager_config` to begin with. I had to add a little extra surface here, since there were a few other places in unrelated API-layer code that directly interacted with different functionality here.

Here's where I differed a bit more I think - since this config api is now part of `notifier`, I was able to reduce the overall exported surface by having the refactored logic depend on _non-exported_ interfaces. This accomplishes the core request of not directly depending on _all_ of MultiOrgAlertmanager, but still avoids expanding the API exported by `notifier` so much. I'm curious as to your thoughts, now that the consuming logic is part of the same package defining all these types? Does this fix your concerns, or is there a different approach I should have taken?